packages/renderer/src/lib/deployments/DeploymentsList.svelte

```diff
@@ -30,21 +29,23 @@ import DeploymentColumnStatus from './DeploymentColumnStatus.svelte';
 import DeploymentEmptyScreen from './DeploymentEmptyScreen.svelte';
 import type { DeploymentUI } from './DeploymentUI';
 
-export let searchTerm = '';
-$: deploymentSearchPattern.set(searchTerm);
-
-let deployments: DeploymentUI[] = [];
+interface Props {
+  searchTerm?: string;
+}
 
-const deploymentUtils = new DeploymentUtils();
+let { searchTerm = '' }: Props = $props();
 
-onMount(() => {
-  return kubernetesCurrentContextDeploymentsFiltered.subscribe(value => {
-    deployments = value.map(deployment => deploymentUtils.getDeploymentUI(deployment));
-  });
+$effect(() => {
+  deploymentSearchPattern.set(searchTerm);
 });
 
+const deploymentUtils = new DeploymentUtils();
+const deployments = $derived(
+  $kubernetesCurrentContextDeploymentsFiltered.map(deployment => deploymentUtils.getDeploymentUI(deployment)),
+);
+
```

Comment:
AFAIK derived objects from $store pattern are not refreshed when the store is updated

Reply:
It seems that it works, as the deployments are refreshed when the search term is modified. Also, this outputs logs:

```
const deployments = $derived.by(() => {
    console.log('==> derive');
    return $kubernetesCurrentContextDeploymentsFiltered.map(deployment => deploymentUtils.getDeploymentUI(deployment));
  }
);
```

Reply:
Yes, I understand. When searchTerm is modified, kubernetesCurrentContextDeploymentsFiltered is updated, and I can see that the deployments list is modified.

Another example where `$derived($store)` seems to work correctly: https://svelte.dev/playground/a686354d4c9f4ce0bdb8ba63672ff775?version=5.19.1

Or am I missing something?

Reply:
> AFAIK derived objects from $store pattern are not refreshed when the store is updated

On my side it seems it is also refreshed on other example

Reply:
good to know, it must have been fixed in some new updates of svelte v5

please add a test where kubernetesCurrentContextDeploymentsFiltered is being updated while the component is already rendered initially to ensure we'll have no regression on that part

Reply:
done, I had to refactor the unit tests, to mock the stores directly, to be able to add this test